### PR TITLE
Implement Annual Spending View

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context
**Type:** Feat  
**Intent:** Implement a complete Annual Spending View to let users visualize and analyze spending aggregated by bill for a chosen year. Adds server-side aggregation, validation, UI pages/components (chart, list, summary, navigation), and related docs/tests.

### 🧭 Reviewer Guide
**Complexity:** Medium

#### Entry Point
Start with `lib/types.ts` to understand the new shapes: `AggregatedBillSpending` and `AnnualSpendingSummary`. Then inspect `lib/services/TransactionService.ts` for the new aggregation method `getPaymentsByYearAggregatedByBill(year: string)` which contains the core query and grouping logic. Next, review `actions/history.ts` for `annualSpendingQuerySchema` and `getAnnualSpendingData()` (validation, orchestration, and error logging). Finally, review the page and component flow: `app/annual-spending/page.tsx` → `components/features/history/AnnualSpendingContent.tsx` → interactive UI pieces (`AnnualSpendingInteractive`, `AnnualSpendingGraph`, `AnnualSpendingChart`, `AnnualSpendingList`, `AnnualSpendingRow`, `AnnualSpendingSummary`, `AnnualSpendingSkeleton`) and `lib/services/HistoryService.ts` for `calculateAnnualSummary()`.

#### Sensitive Areas
- `lib/services/TransactionService.ts` - SQL/ORM query building and aggregation: verify date range filtering for full calendar year, correct grouping by bill, correct computation of `paymentCount`, `totalAmount`, `averageAmount`, sorting by `totalAmount`, and numeric rounding.
- `actions/history.ts` - Strict year validation (YYYY regex) and ActionResult error handling/logging paths.
- `lib/search-params.ts` - `parseAsYear`, `annualSpendingSearchParams`, and `annualSpendingSearchParamsCache`: ensure default/current-year behavior and server-side caching correctness.
- `lib/services/HistoryService.ts` - `calculateAnnualSummary()` aggregation correctness (total bills, total payments, amount paid).
- `components/features/history/AnnualSpendingChart.tsx` - Client-side chart rendering, color mapping, sanitize billId → color key, click handling and tooltip formatting.
- `components/features/history/AnnualSpendingInteractive.tsx` - State synchronization between chart and list (highlight toggling).
- `app/annual-spending/page.tsx` & Suspense usage - Server component concurrency (settings, tags, categorized data) and correct fallback with `AnnualSpendingSkeleton`.
- Tests added: `actions/history.test.ts`, `lib/services/TransactionService.test.ts`, `lib/services/HistoryService.test.ts` — verify test coverage for validation, aggregation logic, edge cases, and error paths.

### ⚠️ Risk Assessment
- **Breaking Changes:** No breaking changes - all modifications add new types, services, pages, components, docs, and tests; existing public APIs are extended but not altered in incompatible ways.  
- **Migrations/State:** No migrations or state changes

### CODE REVIEW

File: components/features/history/AnnualSpendingContent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      data-testid="annual-spending-interactive"
+      data-currency={currency}
+      data-locale={locale}
+      data-year={year}
+      data-bill-count={Array.isArray(data) ? data.length : 0}
+    >
+      Interactive
+    </div>
+  ),
+}));
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingContent.test.tsx]: I prefer leave this as is for now

File: components/features/history/AnnualSpendingGraph.test.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      data-testid="annual-spending-chart"
+      data-currency={currency}
+      data-locale={locale}
+      data-highlighted-bill-id={highlightedBillId}
+      data-bill-count={data.length}
+    >
+      Chart
+    </div>
+  ),
+}));
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingGraph.test.tsx]: I prefer leave this as is for now

File: components/features/history/AnnualSpendingList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+          currency="USD"
+          locale="en-US"
+          year="2025"
+        />
+      );
+
+      expect(screen.getByTestId('row-bill-1')).toBeInTheDocument();
+      expect(screen.queryByTestId('row-bill-2')).not.toBeInTheDocument();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingList.test.tsx]: I prefer leave this as is for now

File: components/features/history/AnnualSpendingRow.test.tsx [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+            />
+          </tbody>
+        </table>
+      );
+
+      expect(formatMoney).toHaveBeenCalledWith(99999999, 'USD', 'en-US');
+      expect(formatMoney).toHaveBeenCalledWith(8333333, 'USD', 'en-US');
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingRow.test.tsx]: This one should be fixed now. Please recheck

File: components/features/history/AnnualSpendingList.test.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingList.test.tsx]: This one should be fixed now. Please recheck

File: components/features/history/AnnualSpendingList.test.tsx [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+          bills={mockBills}
+          currency="PLN"
+          locale="pl-PL"
+          year="2025"
+        />
+      );
+
+      expect(screen.getByTestId('row-bill-1')).toBeInTheDocument();
+      expect(screen.getByTestId('row-bill-2')).toBeInTheDocument();
+    });
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/AnnualSpendingList.test.tsx]: Ok, I  just deleted this test since the actual row component tests already verify formatting works correctly.
